db/products.js

```diff
@@ -0,0 +1,51 @@
+class Products {
+  constructor() {
+    //saving data here
+    //{id: Number, name: String, price: Number, inventory: Number}
+    this._productsArr = [];
+    this._sku = 1;
+  }
+
+  getAllProducts() {
+    return this._productsArr;
+  }
+
+  setId(product) {
+    product.price = product.price * 1;
+    product.inventory = parseInt(product.inventory);
+    product.id = this._sku++;
+    this._productsArr.push(product);
+    console.log(this._productsArr);
+  }
+
+  getProductById(product) {
+    function search(nameKey, myArray){
```

Comment:
You could pull the for loop out of this function and just set a let variable equal to the found product. 

```
let thisProduct; 
for (let i = 0; i < myArray.length: i++) {
  if (myArray[i].name === nameKey) { thisProduct = myArray[i]; }
}
```